clang/lib/Sema/AnalysisBasedWarnings.cpp

```diff
@@ -2551,6 +3807,9 @@ void clang::sema::AnalysisBasedWarnings::IssueWarnings(
                        SourceLocation())) {
     CallableVisitor(CallAnalyzers).TraverseTranslationUnitDecl(TU);
   }
+
+  if (S.Context.hasAnyFunctionEffects())
```

Comment:
This doesn't necessarily need to be here in AnalysisBasedWarnings (though it does borrow a utility function `isNoexcept()`). It's just convenient because this is the right general time, at the end of Sema, to do this check.

Reply:
This does seem like it would be the right time to perform this analysis, yeah.